src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -117,6 +118,14 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
           case sc: SingleConstant => sc.context.typ(sc.idDef)
           case _ => ???
         }
+        case Some(p: ap.GlobalVariable) => p.symb match {
+          case sv: SingleGlobalVariable =>
+            sv.typOpt.map(sv.context.symbType).getOrElse(sv.context.typ(sv.expOpt.get))
+          case mv: MultiGlobalVariable => mv.expOpt.map(mv.context.typ) match {
+            case Some(t: InternalTupleT) => t.ts(mv.idx)
+            case t => violation(s"Expected a tuple, but got $t instead")
```

Comment:
I expect that the other type-checks ensure that this will never go wrong, but I guess we can include this conservative check instead of performing the cast to `InternalTuple` right away.